### PR TITLE
Revisit PR 565

### PR DESCRIPTION
**Pull request by christinaholtNOAA**
**Synopsis**

Applying a few changes requested for PR #565 after it was merged. Specifically, finishing up docs and changing to dot-delimited syntax for the `--output-block` arg.

**Type**

- [ ] Bug fix (corrects a known issue)
- [ ] Code maintenance (refactoring, etc. without behavior change)
- [ ] Documentation
- [x] Enhancement (adds a new functionality)
- [ ] Tooling (CI, code-quality, packaging, revision-control, etc.)

**Impact**

- [x] This is a breaking change (changes existing functionality)
- [ ] This is a non-breaking change (existing functionality continues to work as expected)

**Checklist**

- [x] I have added myself and any co-authors to the PR's _Assignees_ list.
- [x] I have reviewed the documentation and have made any updates necessitated by this change.

### CODE REVIEW

**Comment by christinaholtNOAA** [comment]
We likely need additional attention and discussion on the outstanding point about which types to accept as UW YAML keys. I have not (yet) addressed that here.

File: src/uwtools/api/config.py [hunk 1]
```diff
@@ -212,6 +212,7 @@ def _ensure_config_arg_type(
 
 :param input_config: Input config file (``None`` or unspecified => read ``stdin``)
 :param input_format: Format of the input config (optional if file's extension is recognized)
+:param output_block: Path through keys to the desired output block
```
Comment (maddenp-noaa) [inline src/uwtools/api/config.py:215]: I think `uwtools.config.tools.realize_config()` needs a docstring update, too.

Comment (christinaholtNOAA) [reply src/uwtools/api/config.py:215]: Nice catch!

**Review by WeirAE — APPROVED** [review]
With the other doc string, LGTM!

**Review by maddenp-noaa — APPROVED** [review]

**Pull request merged**